wordpress-mobile/WordPress-iOS
# WeeklyRoundupBackgroundTask - Blog Title + Site Title tests and refactor

This PR refactors the Blog+Title.swift and title logic and WeeklyRoundupBackgroundTask notificationTitle logic and adds Unit tests for both

Ref #17956

To test:
1. Test on device.  When app launches go to Me -> App Settings -> Debug.  Tap Weekly Roundup
2. If you don't have enough blogs / history also slide to the right to "Include A8c P2s"
3. Tap one of the Schedule in 10 sec options and then minimise the app in background
4. Weekly Round up Notification should eventually appear per iOS scheduling of BackgroundTask
5. Also run new unit tests:
	1. BlogTitleTests.swift
	2. WeeklyRoundupBackgroundTaskTests.swift

There should be no failures


## Regression Notes
1. Potential unintended areas of impact
WeeklyRoundupBackgroundTask

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I've added unit tests + tested on device

3. What automated tests I added (or what prevented me from doing so)
 * BlogTitleTests.swift
 * WeeklyRoundupBackgroundTaskTests.swift
	
PR submission checklist:
